app/components/SearchableMetricSelector.js

```diff
@@ -1,25 +1,62 @@
 import { useEffect, useState } from 'react'
 
+function SortDictionary(dictionary){
+    var sortedTagsArr = Object.entries(dictionary).sort((a,b) => b[1].length - a[1].length)
+    var retArr = [];
+    for(let i = 0; i < sortedTagsArr.length; i++){
+      var tempArr = sortedTagsArr[i]
+      var tempDict = {};
+      tempDict[tempArr[0]] = tempArr[1]
+      retArr.push(tempDict)
+    }
+    return retArr
+  }
+  
+function CreateTagDictionary(supportedMetrics){
+  const groups = [...new Set(Object.keys(supportedMetrics).map(q => supportedMetrics[q].dataset))];
+  const metricRecords = Object.keys(supportedMetrics)
+  var tagDictionary = {};
+  
+  for (var i in groups){
+    var groupedRecords = [];
+    for(var j in Object.keys(supportedMetrics)){
+      var record = Object.keys(supportedMetrics)[j]
+      if(groups[i].indexOf(supportedMetrics[record].dataset) > -1 ){
+        groupedRecords.push(metricRecords[j])
+      }
+    }
+    tagDictionary[groups[i]] = groupedRecords;
+  }
+  
+  return tagDictionary
+}
+
+function CreateDictArray(tagDictionary){
+  var tagDictArray = [];
+  var chunk_size = 1;
+  for ( var cols = Object.entries( tagDictionary ); cols.length; ){
+    tagDictArray.push( cols.splice(0, chunk_size).reduce( (o,[k,v])=>(o[k]=v,o), {}));
```

Comment:
Can you rewrite this to make it more readable? It also looks like it's copied from somewhere, so either add a link to the source or rewrite it in a way that you understand and think will be readable to others on our team.

Reply:
@fabrego524 Did you see this comment?